mithril-common/src/chain_observer/pallas_observer.rs

```diff
@@ -158,13 +245,46 @@ impl ChainObserver for PallasChainObserver {
 mod tests {
     use std::fs;
 
-    use pallas_codec::utils::AnyCbor;
-    use pallas_network::miniprotocols::localstate::{self, ClientQueryRequest};
+    use pallas_codec::utils::{AnyCbor, AnyUInt, KeyValuePairs, TagWrap};
+    use pallas_crypto::hash::Hash;
+    use pallas_network::miniprotocols::localstate::{self, queries_v16::Value, ClientQueryRequest};
     use tokio::net::UnixListener;
 
     use super::*;
     use crate::{chain_observer::test_cli_runner::TestCliRunner, CardanoNetwork};
 
+    fn get_utxo_mock() -> UTxOByAddress {
```

Comment:
This could be named differently as it is not really a mock:
```suggestion
    fn get_fake_utxo_by_address() -> UTxOByAddress {
```